phoenix-core/src/it/java/org/apache/phoenix/end2end/PhoenixRowTimestampFunctionIT.java

```diff
@@ -191,4 +192,158 @@ public void testRowTimestampColumn() throws Exception {
         }
     }
 
+    @Test
+    public void testRowTimestampColumnAndPredicates() throws Exception {
```

Comment:
all these cases deserve different test functions, please split them. extracting common code will make the test suite more maintainable.